x/mint/abci.go

```diff
@@ -35,7 +35,9 @@ func maybeSetGenesisTime(ctx sdk.Context, k keeper.Keeper) {
 func maybeUpdateMinter(ctx sdk.Context, k keeper.Keeper) {
 	minter := k.GetMinter(ctx)
 	newInflationRate := minter.CalculateInflationRate(ctx)
-	if newInflationRate == minter.InflationRate {
+
+	isNonZeroAnnualProvisions := !minter.AnnualProvisions.IsZero()
+	if newInflationRate.Equal(minter.InflationRate) && isNonZeroAnnualProvisions {
 		// The minter's InflationRate AnnualProvisions already reflect the
 		// values for this year. Exit early because we don't need to update
 		// them.
```

Comment:
the good news is this appears to not break state or the result of the inflation, but previous to this change, this code was literally never getting hit, so we would always end up updating inflation. That might actually still change the apphash even if the state is not changing, depending on if this is causing an insertion to the iavl tree